server/konnectors/linkedin.coffee

```diff
@@ -0,0 +1,340 @@
+CompareContacts = require '../lib/compare_contacts'
+log = require('printit')
+    prefix: "Linkedin"
+    date: true
+
+ACCOUNT_TYPE = 'com.linkedin'
+
+
+
+module.exports =
+
+    name: "Linkedin"
+    slug: "linkedin"
+    description: "konnector description linkedin"
+
+    fields:
+        login: "text"
+        password: "password"
+    models:
+        contact: Contact
+
+    fetch: (requiredFields, callback) ->
+        log.info 'Import started'
+        fetcher.new()
+            .use(retrieveTokens)
+            .use(logIn)
+            .use(retrieveListContact)
+            .use(prepareCozyContacts)
+            .use(retrieveContacts)
+            .args(requiredFields, {}, {})
+            .fetch (err, fields, entries) ->
+                return callback err if err
+                log.info "Import finished"
+                callback()
+
+
+# Load landing page to retrieve the csrf token needed in login request.
+# More info on csrf -> https://en.wikipedia.org/wiki/Cross-site_request_forgery
+#
+# The html parsing is done with cheerio, a "jquery like" server side
+retrieveTokens = (requiredFields, entries, data, next) ->
+    log.info 'Retrieve Tokens'
+    opts =
+        url: 'https://linkedin.com'
+        jar: true
+
+    request.get opts, (err, res, body) ->
+        return next err if err
+
+        if body.status? and body.status is 'error'
+            return next new Error(body.status_details)
+
+        $ = cheerio.load body
+        entries.csrfToken = $('#loginCsrfParam-login').val()
+        entries.accountName = requiredFields.login
+        log.info 'Tokens retrieved'
+        next()
+
+
+
+# Make the login request with the user inputs (login/password) and the csrf
+# token retrieved above.
+logIn = (requiredFields, entries, data, next) ->
+    log.info 'Attempt login'
+    opts =
+        url: 'https://www.linkedin.com/uas/login-submit'
+        jar: true
+        form:
+            session_key: requiredFields.login
+            session_password: requiredFields.password
+            loginCsrfParam: entries.csrfToken
+            submit: "Sign+in"
+
+    request.post opts, (err, res, body) ->
+        return next err if err
+
+        if body is ""
+            log.info 'Login success'
+            next()
+        else
+            log.error 'Bad login / password'
+            next new Error("Bad login/Password")
+
+
+# Retrieve a list of all contacts id available in the count. The id will be
+# use to retrieve his personnal data.
+retrieveListContact = (requiredFields, entries, data, next) ->
+    log.info 'Retrieve list contact'
+
+    contacts = "https://www.linkedin.com/contacts/api/contacts/?start=0&count=10000&fields=id"
+    opts =
+        url: contacts
+        jar: true
+        json: true
+
+    request.get opts, (err, res, body) ->
+        return next err if err
+        if body.status? and body.status is 'error'
+            return next new Error(body.status_details)
```

Comment:
Please do not use return in the middle of function, us `if` ... `else` block instead.